cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1183,6 +1207,11 @@
 (rule (lower (has_type $I128 (bnot x)))
       (i128_not x))
 
+;; f32 and f64
+
+(rule -3 (lower (has_type (ty_scalar_float ty) (bnot x)))
+      (sse_xor ty x (vector_all_ones)))
```

Comment:
I think it might be preferable to use `ANDNPS` here: `AND`-ing the register to itself keeps the same bits in place and the `NOT` part will do what we want it to do. That way we avoid an extra `PCMPEQ*` due to the `vector_all_ones`.

Reply:
I am not sure we have to make mask for xor or and after xor. also why than our eq tests does not fail because of side-effects?

Reply:
Can you re-phrase your question? I'm not sure exactly what you mean...

Reply:
Oh... I misunderstood. just nevermind.

Reply:
I don't think this works. the and`not` will negative the float value itself. here is result.
```
Caused by:
    Failed test: run: %bnot_f32(0.0) == -NaN:0x3fffff, actual: 0.0
```

Reply:
You can check over here https://www.felixcloutier.com/x86/andnps#andnps--128-bit-legacy-sse-version-

Reply:
Oh, you're right! The NOT happens before the AND. Sorry for the distraction, let me take one more look at everything else...